src/main/java/org/elasticsearch/indices/query/IndicesQueriesModule.java

```diff
@@ -30,30 +30,21 @@
 
 public class IndicesQueriesModule extends AbstractModule {
 
-    private Set<Class<QueryParser>> queryParsersClasses = Sets.newHashSet();
-    private Set<QueryParser> queryParsers = Sets.newHashSet();
+    private Set<Class<? extends QueryParser>> queryParsersClasses = Sets.newHashSet();
 
-    public synchronized IndicesQueriesModule addQuery(Class<QueryParser> queryParser) {
+    public synchronized IndicesQueriesModule addQuery(Class<? extends QueryParser> queryParser) {
         queryParsersClasses.add(queryParser);
         return this;
     }
 
-    public synchronized IndicesQueriesModule addQuery(QueryParser queryParser) {
```

Comment:
I mean - global state in a query parser is probably a bad idea so I don't feel bad about you making it harder to do.

I have some vague worries that plugins require both an understanding of guice and Elasticsearch's onModule style initializers which makes plugin development daunting. But I don't have a solution to that. This pull request helps though, so yay!